src/EnergyPlus/InputProcessing/IdfParser.cc

```diff
@@ -198,6 +198,14 @@ json IdfParser::parse_idf(std::string const &idf, size_t &index, bool &success,
         objectTypeMap.emplace(std::move(key), it.key());
     }
 
+    if (idf.size() > 3) {
+        // UTF-8 Byte Order Mark
+        if (idf[0] == '\xEF' && idf[1] == '\xBB' && idf[2] == '\xBF') {
```

Comment:
@mbadams5 do the initial BOM bytes have to be in this order? can they be swapped?

Reply:
That's the [required order](https://en.wikipedia.org/wiki/Byte_order_mark#UTF-8) for a UTF-8 compliant BOM.  Other encodings (UTF-16) have others, but I don't think we need to concern ourselves with those.  Though, @mbadams5, if we are able to read [other encoding types](https://en.wikipedia.org/wiki/Byte_order_mark#Byte_order_marks_by_encoding) that use BOM, would it be worthwhile to add multiple catches here?

Reply:
@Myoldmopar The JSON library only supports UTF-8 so I don't see the benefit of adding the other encodings BOM.

Reply:
The review is complete then.

Reply:
How would you know if an idf was formatted in a non-supported UTF encoding?

Reply:
It would likely fail to parse and if it doesn't then the JSON parser will either reject or ignore the non-supported codepoints.

Reply:
Back to you guys on this one. I just wanted to talk out loud about this to see if something else was needed here. This change allows a step forward to anticipating UTF encoding so at least it's in the right direction. This could go in as-is for now. The CI failure for EnergyPlusFixture.InternalHeatGains_ElectricEquipITE_DefaultCurves doesn't look related.